### PR TITLE
Get Pages through Extension instead of CoreTS directly

### PR DESCRIPTION
**Pull request by streetcoder123**
Task: https://microsoftgarage.visualstudio.com/Intern%20GitHub/_workitems/edit/30277

**Changes**
- Currently, the client directly calls the CoreTS Engine to get pages and put in the wizard Content reducer.
- Now, the client should communicate to the extension that it wants to get pages from CoreTS, extension makes request to the CoreTS and returns the response to the client, which then updates the state.

**How to Check**
- Launch the wizard
- Navigate through it - `Pages` page should be working as usual
- Try to generate a project - Project should be appropriately generated.

### CODE REVIEW

**Comment by SahilTara** [comment]
Think you accidentally added the remove pages stuff in this PR @njindal239

**Comment by streetcoder123** [comment]
@SahilTara Yeah, I don't know how some stuff has been added to this PR. I will rebase this once the other one gets merged.

File: src/client/src/containers/SelectPages/index.tsx [hunk 1]
```diff
@@ -55,40 +55,34 @@ const messages = defineMessages({
...
-      getPages(
-        selectedProjectType.internalName,
-        selectedFrontend.internalName,
-        selectedBackend.internalName,
-        serverPort
-      );
-    }
+    const { selectedBackend, selectedFrontend } = this.props;
+
+    const { vscode } = this.props;
```
Comment (piotrmark) [inline src/client/src/containers/SelectPages/index.tsx]: Can we merge these lines?

File: src/client/src/containers/SelectPages/index.tsx [hunk 2]
```diff
@@ -55,40 +55,34 @@ const messages = defineMessages({
...
-
-      getPages(
-        selectedProjectType.internalName,
-        selectedFrontend.internalName,
-        selectedBackend.internalName,
-        serverPort
-      );
+      const { selectedBackend, selectedFrontend } = this.props;
+
+      const { vscode } = this.props;
```
Comment (piotrmark) [inline src/client/src/containers/SelectPages/index.tsx]: Same here

File: src/extension/src/coreTSModule.ts [hunk 3]
```diff
@@ -3,31 +3,49 @@ import { ExtensionCommand } from "./constants";
...
-        return {
-            payload: {
-                frameworks: result,
-                isPreview: message.payload.isPreview,
-                projectType: message.payload.projectType
-            }
-        };
-    }
+  async getFrameworks(message: any): Promise<IPayloadResponse> {
+    let result = await CoreTemplateStudio.GetExistingInstance().getFrameworks(
```
Comment (jasmingoh) [inline src/extension/src/coreTSModule.ts]: I know you didn't write this, but can this be changed to const?

Comment (streetcoder123) [reply src/extension/src/coreTSModule.ts]: Definitely!

File: src/client/src/containers/SelectPages/index.tsx (hunk 1 above)
Comment (streetcoder123) [reply src/client/src/containers/SelectPages/index.tsx]: Oops, thanks for that :p

**Comment by jcoc611-microsoft** [comment]
After this work can we delete this file now? https://github.com/microsoft/WebTemplateStudio/blob/dev/src/client/src/services/EngineAPIService.tsx

**Comment by jasmingoh** [comment]
@jcoc611-microsoft I'll be doing the clean up in another PR

**Review by jasmingoh — APPROVED** [review]

**Review by SahilTara — APPROVED** [review]

**Review by jcoc611-microsoft — APPROVED** [review]

**Review by jsondoo — APPROVED** [review]
ship it

**Pull request merged**